community/detectors/argocd_exposed_ui/src/main/java/com/google/tsunami/plugins/detectors/exposedui/argocd/ExposedArgoCdApiDetector.java

```diff
@@ -0,0 +1,452 @@
+                        Severity.HIGH));
+              } else if (isServiceVulnerableToAuthBypass(networkService, false)) {
+                // Argo CD API server is vulnerable to CVE-2022-29165, and it is
+                // confirmed by receiving matching a http response body
+                detectionReport.addDetectionReports(
+                    buildDetectionReport(
+                        targetInfo,
+                        networkService,
+                        "Argo CD API server is vulnerable to CVE-2022-29165. "
+                            + "The authentication can be bypassed. "
+                            + "We can't confirm that this API server has an admin role because we "
+                            + "can't create a new application and receive an out-of-band callback from it, "
+                            + "but we are able to receive some endpoint data without authentication",
+                        "Patched versions are 2.1.15, and 2.3.4, and 2.2.9, and"
+                            + " 2.1.15. Please update Argo CD to these versions and higher.",
+                        Severity.HIGH));
+              }
+            });
+    return detectionReport.build();
+  }
+
+  private boolean isInScopeService(NetworkService networkService) {
+    return NetworkServiceUtils.isWebService(networkService)
+        || HTTP_EQUIVALENT_SERVICE_NAMES.contains(networkService.getServiceName());
+  }
+
+  /** Checks if a {@link NetworkService} has a misconfigured ArgoCD API server exposed. */
+  private boolean isServicePubliclyExposed(
+      NetworkService networkService, boolean useOutOfBandCallBack) {
+    if (useOutOfBandCallBack) {
+      return checkExposedArgoCdWithOutOfBandCallback(networkService, HttpHeaders.builder());
+    } else {
+      return checkExposedArgoCdWithResponseMatching(networkService, HttpHeaders.builder());
+    }
+  }
+
+  /** Checks if a {@link NetworkService} has a vulnerable ArgoCD API server to CVE-2022-29165. */
+  private boolean isServiceVulnerableToAuthBypass(
+      NetworkService networkService, boolean useOutOfBandCallBack) {
+    HttpHeaders.Builder cookieHeader =
+        HttpHeaders.builder().addHeader("Cookie", PAYLOAD_ARGOCD_TOKEN_SESSION);
+    if (useOutOfBandCallBack) {
+      return checkExposedArgoCdWithOutOfBandCallback(networkService, cookieHeader);
+    } else {
+      return checkExposedArgoCdWithResponseMatching(networkService, cookieHeader);
+    }
+  }
+
+  private boolean checkExposedArgoCdWithResponseMatching(
+      NetworkService networkService, HttpHeaders.Builder baseHeaders) {
+    logger.atInfo().log("Starting exposed Argo CD API servers detection by response matching.");
+    // the target URL of the target is built
+    String targetUrl = NetworkServiceUtils.buildWebApplicationRootUrl(networkService);
+
+    String targetUri = targetUrl + "api/v1/certificates";
+    logger.atInfo().log("targetUri is %s", targetUri);
```

Comment:
Consider removing this line, since sent HTTP requests are already logged by default